### PR TITLE
Add b2_index() and tests for tree balance

### PR DESCRIPTION
**Pull request by Billyzhang1229**
Add doc for sackin_index()

### CODE REVIEW

File: phylokit/balance.py [hunk 1]
```diff
@@ -97,3 +108,59 @@ def b1_index(tree):
...
+
+@numba.njit(cache=True)
+def _b2_index(virtual_root, left_child, right_sib, base):
+    root = left_child[virtual_root]
+    stack = [(root, 1)]
+    total_proba = 0.0
+    while len(stack) > 0:
+        u, path_product = stack.pop()
+        if left_child[u] == -1:
+            total_proba -= path_product * general_log(path_product, base)
```
Comment (jeromekelleher) [inline phylokit/balance.py:134]: We can probably just call ``math.log(path_product, base)`` here? I only did the ``general_log`` thing because C doesn't have this function.

**Review by jeromekelleher — COMMENTED** [review]
LGTM.

The git history has got muddled up though, and you need to rebase. [This guide](https://popsim-consortium.github.io/stdpopsim-docs/stable/development.html#github-workflow) that I wrote for another project should help.

If you're getting stuck ping me or Ben on Slack and we'll walk you through it (everyone get's stuck on rebasing the first few times!)

**Comment by jeromekelleher** [comment]
Also, please squash down to one commit while you're rebasing (in general, one commit per PR is a good rule of thumb, unless there's a set of discrete things that would be good to keep history for in there)

File: phylokit/balance.py (hunk 1 above)
Comment (Billyzhang1229) [reply phylokit/balance.py:134]: This is because the base argument is not implemented in Numba. 
In the future, we could probably move `general_log(x, base)` to `core.py` or something for any other potential usage?
```
There are 2 candidate implementations:
        - Of which 2 did not match due to:
        Type Restricted Function in function 'log': File: unknown: Line unknown.
          With argument(s): '(float64, float64)':
         No match for registered cases:
          * (int64,) -> float64
          * (uint64,) -> float64
          * (float32,) -> float32
          * (float64,) -> float64
```

Comment (jeromekelleher) [reply phylokit/balance.py:134]: Aha, fair enough. All good then, let's leave it as it is. Let's not bother putting it into another file yet as we'd have to do more testing of the numerical corner cases.

**Comment by Billyzhang1229** [comment]
Rebased✅

**Review by jeromekelleher — APPROVED** [review]
